apps/meteor/app/livechat/server/lib/parseTranscriptRequest.ts

```diff
@@ -0,0 +1,59 @@
+import type { ILivechatVisitor, IOmnichannelRoom, IUser } from '@rocket.chat/core-typings';
+import { LivechatVisitors, Users } from '@rocket.chat/models';
+
+import { settings } from '../../../settings/server';
+import type { CloseRoomParams } from './localTypes';
+
+export const parseTranscriptRequest = async (
+	room: IOmnichannelRoom,
+	options: CloseRoomParams['options'],
+	visitor?: ILivechatVisitor,
+	user?: IUser,
+): Promise<CloseRoomParams['options']> => {
+	const visitorDecideTranscript = settings.get<boolean>('Livechat_enable_transcript');
+	// visitor decides, no changes
+	if (visitorDecideTranscript) {
+		return options;
+	}
+
+	// send always is disabled, no changes
+	const sendAlways = settings.get<boolean>('Livechat_transcript_send_always');
+	if (!sendAlways) {
+		return options;
+	}
+
+	const visitorData = visitor || (await LivechatVisitors.findOneById(room.v._id));
```

Comment:
```suggestion
	const visitorData = visitor || (await LivechatVisitors.findOneById(room.v._id, { projection: { visitorEmails: 1 }));
```
I believe we're missing a `projection` here